### ISSUE TITLE
Acceptance test is successful even when an endpoint cannot be stopped

### ISSUE THREAD

**Issue by danielmarbach** [Particular/NServiceBus#4331]
When an endpoint is stopped in the `ScenarioRunner` and it throws an exception while stopping the task catches the exception and wraps it into a[ `ScenarioException` while rethrowing](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs#L327). The [`Task.WhenAll`](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs#L331) is completed when all the tasks are either completed, cancelled or faulted. But we never actually query the state of the task representing the `WhenAll` when it is returned from [`WhenAny`](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs#L333) when shutting down. 

Sample to reproduce

```
            var tasks = Enumerable.Range(0, 4).Select(async endpoint =>
            {
                Console.WriteLine("Stopping endpoint: {0}", endpoint);
                await Task.Delay(1);
                if (endpoint == 2)
                {
                    throw new Exception("Endpoint failed to stop");
                }
                Console.WriteLine("Endpoint: {0} stopped", endpoint);
            });

            var whenAll = Task.WhenAll(tasks);
            var timeoutTask = Task.Delay(TimeSpan.FromSeconds(5));
            var completedTask = await Task.WhenAny(whenAll, timeoutTask).ConfigureAwait(false);

            if (completedTask == timeoutTask)
            {
                throw new Exception("Stopping endpoints took longer than 2 minutes");
            }
```

This code would be successful even though endpoint 2 threw an exception. Under the assumption it is enough to rethrow the first exception we saw we would need add a single line to the above code snippet:

```
            // as above
            if (completedTask == timeoutTask)
            {
                throw new Exception("Stopping endpoints took longer than 2 minutes");
            }

            await completedTask.ConfigureAwait(false);
```

What should the excepted scenario shutdown be?

**Comment by timbussmann** [comment]
👍 to await the finished task to throw exceptions.

**Comment by timbussmann** [comment]
@danielmarbach do you want to drive this forward?

**Comment by danielmarbach** [comment]
This cannot be fixed before we fix #4332 

It can be easily reproduced with `When_feature_startup_task_fails`:
1. Add the await statement as described above
1. Execute `When_feature_startup_task_fails`

outcome the stop endpoints will bomb with a `NullReferenceException`
